src/main/java/com/product/service/ProductServiceImpl.java

```diff
@@ -0,0 +1,63 @@
+package com.product.service;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import com.product.dao.ProductDaoImpl;
+import com.product.model.Product;
+
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+@Data
```

Comment:
This is Service class, you dont need it to have `@Data` (Service class should never have @data)

Reply:
Done.